app/src/main/java/my/noveldokusha/scraper/utils.kt

```diff
@@ -42,14 +42,19 @@ suspend fun connect(url: String): Connection = Jsoup.connect(url).apply {
     ignoreHttpErrors(true)
 }
 
+// pasre if connect success or return empty list
+fun connect.parseOrEmpty(): Document? = if (this.statusCode == 200) this.parse() else listOf()
+
```

Comment:
Ignoring errors is not  solution for this case

Reply:
currently in class source i am using statuscode manual checking. adding ignoreHttpErrors needed, so that jsoup doesn't throw an error (HttpStatusException).
I only propose that it be used as an extension. so afrer call connect we call parseOrEmpty if responcode 404 is return listOf()
or make it global in the tryConnect function, for this suggestion the ignoring error in jsoup should be omitted.

Reply:
In that case make it internal to that class only and catch the 404 case, then treat the 404 it as an empty result.  But only make it for that source class.

It's true we could keep an history of previous results to know how many pages we have but that would complicate other parts of the code, this current form hides the complexity of each source to its own implementation, making the general interface easier to use and consume. 

Your initial idea is good, just make it internal to only the source.